sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceRDD.scala

```diff
@@ -56,6 +58,15 @@ class DataSourceRDD(
     context.addTaskCompletionListener[Unit](_ => reader.close())
     val iter = new Iterator[Any] {
       private[this] var valuePrepared = false
+      private val inputMetrics = context.taskMetrics().inputMetrics
+      private val existingBytesRead = inputMetrics.bytesRead
```

Comment:
I'm not sure if we can support this if the data source don't report the size metrics. AFAIK ds v1 doesn't support it either. We only support it in file source.

I think we can only support the "recordsRead" metrics for now. We need to design a general API for data sources to report metrics. cc @rdblue @brkyvz

Reply:
ping @rdblue @brkyvz

Reply:
+1 @cloud-fan's advice.

Reply:
if we update only "recordsRead". In the UI Input "Size/Records" column will look like "0/ 10". Is this okay ?

Reply:
that's true for DS v1 too, right?

Reply:
yes it's true for DS V1 also. Shall we have some marker trait which says Input Size metrics is supported. For e.g.  FilePartitionReaderFactory supports Input Size metrics. So that we can collect input size metrics only if it supports .

Reply:
yes we should, but then it's an API design and we should involve more people to dicuss.

Reply:
then shall I update the solution and push, then people can comment on the approach ?

Reply:
I'd suggest we send an email to dev list with your proposal to trigger discussion first. AFAIK many people are interested in the metrics API in DS v2.